### PR TITLE
Add JobStatus

### PR DESCRIPTION
**Pull request by SkyToGround**
First PR. Adds two classes for keeping track of (write) jobs.

### CODE REVIEW

File: file_writer_control/JobStatus.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        self._service_id = ""
+        self._last_update = datetime.now()
+        self._state = JobState.WAITING
+        self._message = ""
+
+    def update_status(self, new_status):
+        """
+        Updates the status/state of a this instance of the JobStatus class, using another instance.
+        .. note:: The job identifier of this instance and the other must be identical.
+        :param new_status: The other instance of the JobStatus class.
```
Comment (umithardal) [inline file_writer_control/JobStatus.py:33]: Is this our convention for parameter doc-strings?  I mean I prefer:

`def foo(x):
       Args:
             x: bla bla
`
Is there a specific reason? There is nothing wrong with it of course.

File: file_writer_control/JobStatus.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        """
+        if new_status.job_id != self.job_id:
+            raise RuntimeError(
+                "Job id of status update is not correct ({} vs {})".format(
```
Comment (umithardal) [inline file_writer_control/JobStatus.py:37]: Instead of `.format()` may be f-strings should be used.

File: file_writer_control/JobStatus.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                )
+            )
+        self._state = new_status.state
+        if new_status.message:
+            self._message = new_status.message
+        self._service_id = new_status.service_id
+        self._last_update = new_status.last_update
+
+    @property
+    def job_id(self) -> str:
```
Comment (umithardal) [inline file_writer_control/JobStatus.py:48]: While this function has an explicit return type others have not. May be it is better to have a coherent codebase. Either we should have a typed code all together or not.

File: file_writer_control/JobStatus.py (hunk 2 above)
Comment (SkyToGround) [reply file_writer_control/JobStatus.py]: I am/was not familiar with f-strings but it has now been changed.

File: file_writer_control/JobStatus.py (hunk 3 above)
Comment (SkyToGround) [reply file_writer_control/JobStatus.py:45]: I did not care to set the return type of functions that did not return anything. Nevertheless, the return type of those functions have now been set to `None`.

File: file_writer_control/JobStatus.py (hunk 1 above)
Comment (SkyToGround) [reply file_writer_control/JobStatus.py:33]: I code too little in Python to have a strong preference yet but looking at some of our other Python projects ([e.g. just-bin-it](https://github.com/ess-dmsc/just-bin-it/blob/master/just_bin_it/endpoints/config_listener.py)), this is the convention that seems to be in use.

Comment (mattclarke) [reply file_writer_control/JobStatus.py:33]: Yes, this is our convention. It is still the most common way of doing docstrings in the Python eco-system as it comes from Sphinx.

Also, PyCharm does it out of the box!

Comment (mattclarke) [reply file_writer_control/JobStatus.py:33]: Not out of the box, you have to change one setting ;)

File: file_writer_control/JobStatus.py (hunk 3 above)
Comment (umithardal) [reply file_writer_control/JobStatus.py:45]: Oh sorry ı was not entirely clear. Sure, if it obviously return None, we do not need it. But for example in update_status method the type of the new_status is still missing. That was what I meant, if we are going for typed, everything should be typed :)

**Review by umithardal — APPROVED** [review]

File: file_writer_control/JobStatus.py (hunk 3 above)
Comment (SkyToGround) [reply file_writer_control/JobStatus.py:45]: Ah, the Python interpreter does not accept type names of the enclosing class. However, it does accept strings it seems so I have used that now.

**Pull request merged**